dll/win32/shell32/dialogs/filetypes.cpp

```diff
@@ -1462,129 +1667,166 @@ EditTypeDlgProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
-    dwStyle = dwStyle | LVS_EX_FULLROWSELECT;
-    SendMessage(hListView, LVM_SETEXTENDEDLISTVIEWSTYLE, 0, dwStyle);
+    const UINT lvexstyle = LVS_EX_FULLROWSELECT | LVS_EX_LABELTIP;
+    ListView_SetExtendedListViewStyleEx(hListView, lvexstyle, lvexstyle);
 }
 
-static PFILE_TYPE_ENTRY
-FileTypesDlg_DoList(HWND hwndDlg)
+static void
+FileTypesDlg_SetGroupboxText(HWND hwndDlg, LPCWSTR Assoc)
 {
-    HWND hListView;
-    DWORD dwIndex = 0;
-    WCHAR szName[50];
-    WCHAR szFile[100];
-    DWORD dwName;
-    LVITEMW lvItem;
-    INT iItem = 0;
-    HIMAGELIST himlLarge, himlSmall;
-
-    // create imagelists
-    himlLarge = ImageList_Create(GetSystemMetrics(SM_CXICON), GetSystemMetrics(SM_CYICON),
-                                 ILC_COLOR32 | ILC_MASK, 256, 20);
-    himlSmall = ImageList_Create(GetSystemMetrics(SM_CXSMICON), GetSystemMetrics(SM_CYSMICON),
-                                 ILC_COLOR32 | ILC_MASK, 256, 20);
-
-    // set imagelists to listview.
-    hListView = GetDlgItem(hwndDlg, IDC_FILETYPES_LISTVIEW);
-    ListView_SetImageList(hListView, himlLarge, LVSIL_NORMAL);
-    ListView_SetImageList(hListView, himlSmall, LVSIL_SMALL);
-
-    FileTypesDlg_InitListView(hwndDlg, hListView);
-
-    szFile[0] = 0;
-    if (!LoadStringW(shell32_hInstance, IDS_FILE_EXT_TYPE, szFile, _countof(szFile)))
-    {
-        // default to english
-        wcscpy(szFile, L"%s File");
-    }
-    szFile[(_countof(szFile)) - 1] = 0;
+    CStringW buf;
+    buf.Format(IDS_FILE_DETAILS, Assoc);
+    SetDlgItemTextW(hwndDlg, IDC_FILETYPES_DETAILS_GROUPBOX, buf.GetString());
+}
 
-    dwName = _countof(szName);
+static void
+FileTypesDlg_Refresh(HWND hwndDlg, HWND hListView, PFILE_TYPE_GLOBALS g)
+{
+    ListView_DeleteAllItems(hListView);
+    ImageList_RemoveAll(g->himlSmall);
+    InitializeDefaultIcons(g);
+    FileTypesDlg_SetGroupboxText(hwndDlg, L"");
+    SetDlgItemTextW(hwndDlg, IDC_FILETYPES_DESCRIPTION, L"");
+    SetDlgItemTextW(hwndDlg, IDC_FILETYPES_APPNAME, L"");
+    EnableWindow(GetDlgItem(hwndDlg, IDC_FILETYPES_DELETE), FALSE);
+    RedrawWindow(hwndDlg, NULL, NULL, RDW_ALLCHILDREN | RDW_INVALIDATE | RDW_UPDATENOW);
+#if DBG
+    DWORD statTickStart = GetTickCount();
+#endif
 
+    INT iItem = 0;
+    WCHAR szName[ASSOC_CCHMAX];
+    DWORD dwName = _countof(szName);
+    DWORD dwIndex = 0;
+    SendMessage(hListView, WM_SETREDRAW, FALSE, 0);
     while (RegEnumKeyExW(HKEY_CLASSES_ROOT, dwIndex++, szName, &dwName,
                          NULL, NULL, NULL, NULL) == ERROR_SUCCESS)
     {
-        if (FileTypesDlg_InsertToLV(hListView, szName, iItem, szFile))
+        if (FileTypesDlg_InsertToLV(hListView, szName, iItem, g))
             ++iItem;
         dwName = _countof(szName);
     }
+    FileTypesDlg_Sort(g, hListView);
+    SendMessage(hListView, WM_SETREDRAW, TRUE, 0);
+    RedrawWindow(hListView, NULL, NULL, RDW_ALLCHILDREN | RDW_ERASE | RDW_FRAME | RDW_INVALIDATE);
+
+#if DBG
```

Comment:
I do. Do you mind?

It is very handy to see how long it takes to load. The whole list loads more than twice as fast now in my VM compared to the old code.

Windows uses a background thread and a waiting animation while we lock the entire UI and this is bad. My average load time is about 500ms which is OK'ish but if this creeps above one second again the whole thing should be reworked to use a worker thread. Any change made in the loading path should be measured.

Reply:
But bear in mind then that this is some code that is going to be currently run everytime one opens the window (my point is that it's going to be active by default in our nightly builds). It should be better to disable it, and only the developer who is working on the code to enable it in his local build.

Reply:
![Dlg](https://github.com/reactos/reactos/assets/138629473/60045bc2-909f-45eb-b79e-e3633eab1271)

Reply:
> this is some code that is going to be currently run everytime one opens the window

The overhead is GetTickCount*2, wsprintf, CreateWindow and SetTimer, it does not slow down the page at all.

Reply:
Please instead of displaying the opening time in the dialog, instead just write it into the log. That is as good as your solution for devs, and will not disturb our "end-users". Keep in mind that even the release-branches currently have to be built in dbg-cfg!